test/BoardTests.cpp

```diff
@@ -70,10 +70,147 @@ TEST_F(BoardTests, checkBoardArgumentBoard_WithGetBoardLinesFunction)
     Lines vec2 = {vec, vec};
 
     Board board(3, 3, vec2, vec2);
-    ASSERT_EQ(board.getBoardLines().at(1)[1],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(1)[0],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(0)[1],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(0)[0],BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1)[1], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1)[0], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[1], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[0], BoardFields::UNKNOWN);
+}
+
+TEST_F(BoardTests, checkBoardSetField)
+{
+    Board board(3, 2, {}, {});
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(1, 0, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getBoardLines().at(0).at(0), BoardFields::BLACK);
+    ASSERT_EQ(board.getBoardLines().at(1).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(2).at(0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1).at(0), BoardFields::WHITE);
+    ASSERT_EQ(board.getBoardLines().at(2).at(1), BoardFields::UNKNOWN);
+
+    board.setField(0, 0, BoardFields::UNKNOWN);
+    board.setField(2, 0, BoardFields::BLACK);
+    board.setField(1, 0, BoardFields::UNKNOWN);
+    board.setField(2, 1, BoardFields::WHITE);
```

Comment:
setting fields is aldready tested above

Reply:
I use it here to to set fields instead of using at(), because they are tested earlier and sure to work correctly.